src/collective.jl

```diff
@@ -39,6 +39,19 @@ function Bcast!(data, root::Integer, comm::Comm)
     Bcast!(Buffer(data), root, comm)
 end
 
+"""
+    Bcast(obj::T, root::Integer, comm::Comm) where T
+
+Broadcast the `obj` from `root` to all processes in `comm`. Returns the object.
+"""
+function Bcast(obj::T, root::Integer, comm::Comm) where T
+    buf = Ref{T}(obj)
+    Bcast!(buf, root, comm)
+    return buf[]
+end
+Bcast(arr::AbstractArray, root::Integer, comm::Comm) = Bcast!(copy(arr), root, comm)
```

Comment:
I am skeptical of the benefit of these copying definitions. In general this is a bit awkward since it requires the user to allocate a consistent `arr` across all ranks and then they don't get a benefit out of having done that allocation.

`bcast` below you can at least use as:

```
obj = Comm_rank(comm) == root ? bcast(obj, root, comm) : bcast(nothing, root, comm)
```

Reply:
I think this is my main reluctance as well: `Bcast!(copy(arr), ...)` doesn't seem that much less useful.

Reply:
Agreed, let's drop this. My main use case is easy broadcasting of scalars.